app/src/main/java/com/materiiapps/gloom/ui/GloomActivity.kt

```diff
@@ -94,20 +100,48 @@ class GloomActivity : ComponentActivity() {
 
     override fun onNewIntent(intent: Intent?) {
         super.onNewIntent(intent)
+        this.intent = intent
+        println("Activity re-entered")
         intent?.let {
-            if (it.data.toString().startsWith("github://com.github.android/oauth")) {
-                println(it.data.toString())
-                it.data!!.getQueryParameter("code")?.let {
-                    lifecycleScope.launch(Dispatchers.IO) {
+            if (it.isOAuthUri()) {
+                if (auth.awaitingAuthType == null) return
+                it.getOAuthCode()?.let {
+                    lifecycleScope.launch {
+                        auth.setAuthState(loading = true)
                         val res = authRepo.getAccessToken(it)
-                        res.ifSuccessful { token ->
-                            auth.authToken = token.accessToken
-                            navigator.replace(RootScreen())
-                        }
+                        res.fold(
+                            success = { token ->
```

Comment:
i would move this to a viewmodel of some sort like mainviewmodel
just for more separation